engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h

```diff
@@ -108,6 +108,8 @@ struct tree_processor {
 			k = std::make_shared<UnionKernel>(expr, kernel_context, query_graph);
 			kernel_context->setKernelId(k->get_id());
 			k->set_type_id(kernel_type::UnionKernel);
+		} else {
```

Comment:
can you make a custom exception that can include the snippet of algebra that coudl not be interpreted and return that exception so we can catch this sepcific problem
that way we can catch BlazingAlgebraNotSupportedException and tell the user what part of the algebra plan was not supported

Reply:
Something like this
`throw std::runtime_error("expression in the Algebra Relational is currently not supported: " + expr);`
OR do you thing another type of exception?

Reply:
like 

```
throw BlazingSqlInvalidAlgebraException("info here")
```

Reply:
dont throw std::runtime_error that can be anything

Reply:
changes were made